### ISSUE TITLE
Cypress not executing any test file first time I open Cypress

### ISSUE THREAD

**Issue by gpenafib** [cypress-io/cypress#1721]
### Current behavior: 
First time I open Cypress, when I try to execute any test file, Cypress loads the Cypress browser window to start executing the test file, but never does. The counter remains counting, but the test file is never executed. 

I get the following error message in the console: "Uncaught Error: runnable must have an id".

If I close the Cypress browser window and run the same test file again or any other test file, Cypress always runs the test file well and with no issue, but not the first time. Why? 

![2018-05-15 08_29_23-e2e-tests](https://user-images.githubusercontent.com/36888026/40041038-fa5b7094-581c-11e8-8734-1d74bce1f255.png)

### Desired behavior:
Cypress runs the test file with no issue and as expected the first time I open Cypress. 

### Steps to reproduce:
In the folder where I have installed Cypress, "Git Bash here", I open Cypress. Then I click on any test file, Cypress opens the browser window to run the test file, but never does. This only happens the first time I open Cypress. If I close the Cypress browser window and I click on the test file again, then it's fine. 

### Versions  
Cypress version: 2.1.0
Operating system: Windows 7
Browser: Chrome

**Comment by jennifer-shehane** [comment]
Does this happen when running the 'example_spec.js' file that was originally seeded? Or is it only on your specific spec files? 

The issue can really only be addressed with a reproducible example, if you can provide anything to help with that.

**Comment by gpenafib** [comment]
Hi @jennifer-shehane , thanks for your message. It does not seem to happen with the 'example_spec.js' file. Is there any way I can attach a file to this issue? you could see my configuration files and one of the files that is failing.

**Comment by gpenafib** [comment]
And do you know what can be causing the error: "Uncaught Error: runnable must have an id"? 
any idea or anything that I could check?

I have added the 'example_spec.js' file to my project. First time I open Cypress and execute this file,  Cypress executes it correctly. It does not happen with other test files, which are not executed first time I open Cypress.

**Comment by gpenafib** [comment]
Hi @brian-mann, thanks for the information. Here you can find the gist I have created for the spec file:
https://gist.github.com/gpenafib/686d66b47a5cbe3a2faccd1d7bca13ec

However, this spec file tests the backend of an application which is currently only deployed in our local machines, so you will not be able to access to the application under test. 

I will try what you suggested.

**Comment by piozaj** [comment]
+1

**Comment by gpenafib** [comment]
Hi @piozaj, are you experiencing the same issue?

**Comment by gpenafib** [comment]
I have not been able to solve the issue yet. Indications given in this issue by the Cypress team have not fixed the issue yet.

**Comment by gpenafib** [comment]
Hi guys, I think setting the base URLs as environment variables in cypress.json could help fixing this issue. So far so good, but I need to keep trying to see if I am able to reproduce this issue. 
Do you think that setting the base URLs as environment variables is related to this issue?

**Comment by zzeni** [comment]
Hi all. I just got the same error. Eventually, it appeared that I had a mistake in my code (I was trying to use an undeclared variable - just some typo - in a function call) which was supposed to throw an exception. It seems that the runner fails to handle the original error, due to an exception in the exception handler itself...

**Comment by jennifer-shehane** [comment]
This error is thrown here in our code: https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cypress/runner.coffee#L779

Unfortunately we'll have to close this issue if no reproducible example is provided. Can anyone provide a way to reproduce this?

**Comment by testquantilope** [comment]
@jennifer-shehane Unfortunately I am not able to provide a reproduction yet, but we are also experiencing this issue since updating cypress to 3.5.0. The issue occurs on test cases that have never showed such behavior before.

It doesn't happen reliably, on one file and for me it doesn't happen on first open, but more often after saving a file. If the issue happens I have to close the browser and restart the test, then it will run through.

I'll check if I can find some more or less reliable reproduction.

**Comment by testquantilope** [comment]
@jennifer-shehane I am sorry, I am unable to find a reliable reproduction. The error is appearing everywhere since 3.5.0.

After 3.6.0 though the error changed a bit. Also I have noticed, that it only happens for us after an error in the application. Interesting here is that I have disabled chromeWebSecurity, so none of these errors should make my tests fail.

Does this help you?
![image](https://user-images.githubusercontent.com/35334883/68332351-a6206980-00a4-11ea-97b8-6bc890428da1.png)
